index.js

```diff
@@ -191,11 +191,13 @@ function fastifyAuth0Verify(instance, options, done) {
     instance.decorateRequest('auth0Verify', auth0Options)
     instance.decorateRequest('jwtDecode', jwtDecode)
 
+    const cache =
+      ttl > 0 ? new NodeCache({ stdTTL: ttl }) : { get: () => undefined, set: () => false, close: () => undefined }
+
     // Create a cache or a fake cache
-    instance.decorateRequest(
-      'auth0VerifySecretsCache',
-      ttl > 0 ? new NodeCache({ stdTTL: ttl }) : { get: () => undefined, set: () => false }
-    )
+    instance.decorateRequest('auth0VerifySecretsCache', cache)
+
+    instance.addHook('onClose', () => cache.close())
```

Comment:
this is the actual fix

Reply:
I must admit that I'm not actually sure that this fixes the issue, because based on the stack trace this is not where the issue is happening, but inside the got package instead. Weird though, because got is not supposed to be using any cache by default, so I'm not even sure why that code it being hit.

Finally, we are using Jest for the plugin's tests too but it's not hanging there.

@PatrickHeneise are you sure that you're not using got directly in your code and using the cache? That may explain it, though I guess you are not doing that.

In any case, this is a good fix even if it doesn't address the issue directly,

@PatrickHeneise can you try this out before we publish a new version?